app/scripts/controllers/incoming-transactions.js

```diff
@@ -161,28 +165,26 @@ export default class IncomingTransactionsController {
     this.store.updateState({
       incomingTxLastFetchedBlocksByNetwork: {
         ...currentBlocksByNetwork,
-        [network]: newLatestBlockHashByNetwork,
+        [CHAIN_ID_TO_TYPE_MAP[chainId]]: newLatestBlockHashByNetwork,
       },
       incomingTransactions: newIncomingTransactions,
     })
   }
 
-  async _fetchAll (address, fromBlock, networkType) {
-    const fetchedTxResponse = await this._fetchTxs(address, fromBlock, networkType)
+  async _fetchAll (address, fromBlock, chainId) {
+    const fetchedTxResponse = await this._fetchTxs(address, fromBlock, chainId)
     return this._processTxFetchResponse(fetchedTxResponse)
   }
 
-  async _fetchTxs (address, fromBlock, networkType) {
+  async _fetchTxs (address, fromBlock, chainId) {
     let etherscanSubdomain = 'api'
-    const currentNetworkID = NETWORK_TYPE_TO_ID_MAP[networkType]?.networkId
 
-    if (!currentNetworkID) {
+    if ([ROPSTEN_CHAIN_ID, RINKEBY_CHAIN_ID, GOERLI_CHAIN_ID, KOVAN_CHAIN_ID].includes(chainId)) {
+      etherscanSubdomain = `api-${CHAIN_ID_TO_TYPE_MAP[chainId]}`
+    } else if (chainId !== MAINNET_CHAIN_ID) {
       return {}
     }
 
```

Comment:
```suggestion
if (!CHAIN_ID_TO_TYPE_MAP[chainId]) {
  return {}
}

const etherscanSubdomain = chainId === MAINNET_CHAIN_ID
  ? 'api'
  : `api-${CHAIN_ID_TO_TYPE_MAP[chainId]}`
```

Reply:
Good idea! Done in e974580030958dd47a632f4eec4214de041ff0b5